core/src/main/java/org/apache/iceberg/util/TableScanUtil.java

```diff
@@ -71,6 +78,57 @@ public static CloseableIterable<FileScanTask> splitFiles(
     return CloseableIterable.combine(splitTasks, tasks);
   }
 
+  public static <T extends FileScanTask> Iterable<CombinedScanTask> planTasks(
+      List<T> splitFiles,
+      long splitSize,
+      int lookback,
+      long openFileCost,
+      StructProjection projection) {
```

Comment:
Why not have it as `StructType`? I believe we just need the type of the key we can't combine by. When we consume this from Spark, we will compute an intersection of all partition specs in `Partitioning` as `StructType`, similar to what we already do in `Partitioning$partitionType` (computes a union of all partition types).

Reply:
You're right. `StructType` is sufficient.